src/opentimelineio/serializableCollection.h

```diff
@@ -60,13 +60,34 @@ class SerializableCollection : public SerializableObjectWithMetadata {
         optional<TimeRange> search_range = nullopt,
         bool shallow_search = false) const;
 
+    // Return the first object that matches the given template type.
+    //
+    // An optional search_range may be provided to limit the search.
+    //
+    // If shallow_search is false, will recurse into children.
+    template<typename T = Composable>
+    Retainer<T> child_if(
+        ErrorStatus* error_status,
+        optional<TimeRange> search_range = nullopt,
+        bool shallow_search = false) const;
+
 protected:
     virtual ~SerializableCollection();
 
     virtual bool read_from(Reader&);
     virtual void write_to(Writer&) const;
 
 private:
+    enum ChildrenIfOptions {
+        match_first = 1,
+        shallow_search = 2
+    };
```

Comment:
This is pretty minor, but I'm used to seeing enums that are meant for bitwise operations expressed with bit shift operators.
```
enum ChildrenIfOptions {
   match_first = 1 << 0,
   shallow_search = 1 << 1
};
```
This may come down to personal preference, but I think it helps to express the intent that the options can be or'ed together, especially when there's only 2 (it's more obvious when the third option has a value of 4).

Reply:
Same for me.

Reply:
There isn't a precedent in OTIO yet on this pattern, we've got one enum declared as int, and the rest are C style, so a minimum edit with the bitshifts is sufficient IMO

Reply:
Sounds good, I'll make the change to the enum.

Overall does this PR seem like a worthwhile change vs. the more specialized function in Roger's PR?

Reply:
Could you drop a link to the more specialized function here so we can see the comparison in context of this thread?

Reply:
https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/998#discussion_r656588870

Reply:
So, does this seem like a good idea? If so I would like to un-draft it, otherwise I'll close it and the corresponding issue:
https://github.com/PixarAnimationStudios/OpenTimelineIO/issues/1001